src/aws-cpp-sdk-core/source/client/ClientConfiguration.cpp

```diff
@@ -81,12 +83,28 @@ void setLegacyClientConfigurationParameters(ClientConfiguration& clientConfig)
         );
 
     if (Aws::Utils::StringUtils::ToLower(useCompressionConfig.c_str())  == "disable") {
-      clientConfig.useRequestCompression = Aws::Client::UseRequestCompression::DISABLE;
+      clientConfig.requestCompressionConfig.useRequestCompression = Aws::Client::UseRequestCompression::DISABLE;
+      AWS_LOGSTREAM_DEBUG(CLIENT_CONFIG_TAG, "Request Compression disabled");
     } else {
       //Using default to true for forward compatibility in case new config is added but SDK is not updated.
-      clientConfig.useRequestCompression = Aws::Client::UseRequestCompression::ENABLE;
+      clientConfig.requestCompressionConfig.useRequestCompression = Aws::Client::UseRequestCompression::ENABLE;
+      AWS_LOGSTREAM_DEBUG(CLIENT_CONFIG_TAG, "Request Compression enabled");
     }
 
+    // Getting min request compression length
+    Aws::String minRequestCompressionString = Aws::Environment::GetEnv(REQUEST_MIN_COMPRESSION_SIZE_BYTES_ENV_VAR);
+    if (minRequestCompressionString.empty())
+    {
+      minRequestCompressionString = Aws::Config::GetCachedConfigValue(REQUEST_MIN_COMPRESSION_SIZE_BYTES_CONFIG_VAR);
+    }
```

Comment:
there is a static method
ClientConfiguration::LoadConfigFromEnvOrProfile that fetches the value from our all config sources

Reply:
I'm using it for the other config option, but for minRequestCompression which is an integer value it wouldn't pass the validator.